packages/server-wallet/src/wallet/store.ts

```diff
@@ -1,14 +1,66 @@
 import Objection from 'objection';
+import {SignedState, calculateChannelId, Objective} from '@statechannels/wallet-core';
 
 import {SignState} from '../protocols/actions';
-import {Channel, SyncState} from '../models/channel';
+import {Channel, SyncState, RequiredColumns} from '../models/channel';
+import {SigningWallet} from '../models/signing-wallet';
+import {logger} from '../logger';
+import {addHash} from '../state-utils';
+import knex from '../db/connection';
 
 export const Store = {
+  transaction: async function(callback: () => Promise<void>): Promise<void> {
+    return knex.transaction(async () => {
+      await callback();
+    });
+  },
   signState: async function(action: SignState, tx: Objection.Transaction): Promise<SyncState> {
     const channel = await Channel.forId(action.channelId, tx);
     const notifications = channel.signAndAdd(action);
     await Channel.query(tx).update(channel);
 
     return notifications;
   },
+
+  addObjective: async function(_objective: Objective): Promise<void> {
+    // TODO: Implement this
+    return Promise.resolve();
+  },
+  addSignedState: async function(signedState: SignedState): Promise<void> {
+    if (!signedState.signatures?.length) {
+      throw new Error('No signature for signed state');
+    }
+
+    const channelId = calculateChannelId(signedState);
+    let channel = await Channel.query()
+      .where('channelId', channelId)
+      .first();
+
+    if (!channel) {
+      const addresses = signedState.participants.map(p => p.signingAddress);
+      const signingWallet = await SigningWallet.query()
+        .whereIn('address', addresses)
+        .first();
+
+      if (!signingWallet) {
+        logger.error(
+          {
+            knownWallets: await SigningWallet.query().select(),
+            addresses,
+          },
+          'Not in channel'
+        );
+        throw Error('Not in channel');
```

Comment:
I'd type this error and place it somewhere. This is a good example of a weird edge case we don't really have documented anywhere yet. `PushMessageError.receivedIrrelevantState` or something like this.

Reply:
Current errors have just been copy and pasted from existing code, so I still need to go thru and clean them up.